src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -286,18 +283,25 @@ const calendarDragAttributes = {
   },
 } as any
 
-const selectedCalendarRange = ref<TimeRange | Date | string>(props.modelValue)
-
-const hasCalendar = computed((): boolean => props.mode !== 'relative')
+// Booleans
+const hasCalendar = computed((): boolean => props.mode !== TimepickerMode.Relative)
+const isSingleDatepicker = computed((): boolean => ModeArrayCustom.includes(props.mode) && !props.range)
 const hasTimePeriods = computed((): boolean => props?.timePeriods?.length > 0)
 const showCalendar = computed((): boolean => state.tabName === 'custom' || !hasTimePeriods.value)
-const submitDisabled = computed((): boolean => {
-  // If either the calendar is in range selection mode, or relative time frames
-  // are present, check whether both `start` and `end` are set;
-  // Otherwise, it's a single date or time, so only check `start`
-  return props.range || hasTimePeriods.value
-    ? !state.selectedRange.start || !state.selectedRange.end
-    : !state.selectedRange.start
+const submitDisabled = ref<boolean>(true)
+
+// Dynamically choose the v-model since the Single date and Date range types are different
+const calendarSingleDate = ref<Date|null>(props.modelValue.start)
+const calendarRange = ref<TimeRange>(props.modelValue)
+const calendarVModel = isSingleDatepicker.value
+  ? calendarSingleDate as DatePickerModel
+  : calendarRange as DatePickerModel
```

Comment:
@adamdehaven dynamic v-model assignment works, i just have to cast either option as `DatePickerModel` :shrug:

Reply:
You might be able to use generics

Reply:
Tried the `const calendarVModel = isSingleDatepicker.value ? ref<Date|null>(props.modelValue.start) : ref<TimeRange>(props.modelValue)`.  Worked well, everything was functional, tests were passing as expected.

However, I did not have the endurance to fight the TS complaints. Got hung up on two type issues I wasn't able to resolve, so I've reverted.

Ready for re-review:
https://github.com/Kong/kongponents/pull/1881/files?w=1